tests/cases/compiler/checkJsdocTypeTagOnExportAssignment2.ts

```diff
@@ -0,0 +1,18 @@
+// @allowJs: true
+// @checkJs: true
+// @outDir: ./out
+// @filename: checkJsdocTypeTagOnExportAssignment2.js
+
+// @Filename: a.ts
+export interface Foo {
+    a: number;
+    b: number;
+}
+
+// @Filename: b.js
+/** @type {import("./a").Foo} */
+export default { c: false };
```

Comment:
can you show that this only allows assignability in one direction with a supertype -> subtype assignment like `{ a: 1 }`?

(it doesn't have to be this test, it could be any of them or a new one)

Reply:
I've added two additional tests (`checkJsdocTypeTagOnExportAssignment5/6.ts`), I hope I understand your point correctly.

Reply:
I meant that I wanted to see an error for assigning `{ a } ==> { a, b }` but not for `{ a, b, c } ==> { a, b }`. Sorry about not being clearer.